subprojects/execution/src/main/java/org/gradle/internal/execution/ExecutionOutcome.java

```diff
@@ -17,7 +17,25 @@
 package org.gradle.internal.execution;
 
 public enum ExecutionOutcome {
+    /**
+     * The outputs haven't been changed, because the work is already up-to-date
+     * (i.e. its inputs and outputs match that of the previous execution in the
+     * same workspace).
+     */
     UP_TO_DATE,
+
+    /**
+     * The outputs of the work have been loaded from the build cache.
+     */
     FROM_CACHE,
-    EXECUTED
+
+    /**
+     * The work has been executed with information about the changes that happened since the previous execution.
+     */
+    EXECUTED_INCREMENTALLY,
+
+    /**
+     * The work has been executed with no incremental change information.
+     */
+    EXECUTED_FULLY
```

Comment:
How about we call this outcome `REBUILT`?

Reply:
How about `EXECUTED`? `REBUILT` implies a previous execution, and I don't like to mix in the verb "build" here.

Reply:
I find it strange that we call the outcome `EXECUTED` and have a `rebuildReason`, `RebuildExecutionStateChanges`, etc. Your point about "rebuild" implying a previous execution applies there as well. I think we should choose one name and stick with it.

So it seems that we use "incremental" to denote the incremental execution. We do that consistently. The question is what is the opposite of "incremental". Using executed here would mean that we use nothing (absence of incremental) to denote the non-incrementalness (that would imply `IncrementalExecutionStateChanges` vs. `DefaultExecutionStateChanges`). We could also use non-incremental to specify the non-incrementalness (e.g. `EXECUTED_NON_INCREMENTALLY`, `NonIncrementalExecutionStateChanges`, etc.).

We could try to use `full` execution as an opposite of incremental as well, though I think we then need to use it consistently.

Currently, I prefer to use non-incremental.

Reply:
`EXECUTED_NON_INCREMENTALLY` it is.